src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocations.Analyzer.cs

```diff
@@ -365,7 +383,61 @@ static bool HasSameParametersPlusCancellationToken(ITypeSymbol cancellationToken
                     }
                 }
 
+                // Overload is  valid if its return type is implicitly convertable
+                var toCompareReturnType = methodToCompareWithAllParameters.ReturnType;
+                var originalReturnType = originalMethodWithAllParameters.ReturnType;
+                if (!compilation.ClassifyCommonConversion(toCompareReturnType, originalReturnType).IsImplicit)
```

Comment:
We have a helper IsAssignableTo in language specific layers that you might be able to use.